File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];
        Arrays.fill(res,1);
        for(int i = 1; i < nums.length;i++){
            res[i] = nums[i-1] * res[i-1];
        }
        int pfix = 1;
        for(int i = res.length-1;i >= 0;i--){
            res[i] *= pfix;
            pfix *= nums[i];
        }
        return res;
    }
}  

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        for(int i =0;i< nums.size();i++){
            for(int j =0;j<nums.size();j++){
                if(nums[i] > nums[j]){
                    swap(nums[i],nums[j]);
                }
            }
        }
        for(int i =0;i<nums.size();i++){
            if(nums[i] == nums[i+1]){
                return true;
            }
        }
        return false;
    }
};
